src/transactions/test/TxEnvelopeTests.cpp

```diff
@@ -582,6 +589,101 @@ TEST_CASE("txenvelope", "[tx][envelope]")
                         });
                     }
 
+                    SECTION("merge source account before payment")
+                    {
+                        auto b1 = root.create("b1", paymentAmount);
+                        a1.pay(b1, 1000);
+
+                        closeLedgerOn(*app, 2, 1, 1, 2016);
+
+                        TransactionFramePtr txMerge;
+
+                        auto setup =
+                            [&](bool txAccountMissing) -> TransactionFramePtr {
+
+                            txMerge = b1.tx({accountMerge(a1)},
+                                            b1.getLastSequenceNumber() + 2);
+
+                            auto& txAccount = txAccountMissing ? b1 : a1;
+                            auto& opAccount = txAccountMissing ? a1 : b1;
+
+                            auto tx =
+                                txAccount.tx({opAccount.op(payment(root, 100)),
+                                              root.op(payment(a1, 100))},
+                                             b1.getLastSequenceNumber() + 3);
```

Comment:
I refactored the tests quite a bit. Also, I set the transaction source account to `txAccount` because I wanted the option to submit a transaction with a missing source account. Is the way I did this wrong?

Reply:
I wasn't quite specific enough in my comment. What you did is right. The thing that confused me was the mismatch between `txAccount` being the source account but the sequence number being based off of `b1` (even though `txAccount` could be `a1` or `b1`). But it worked because your test was written in such a way that `a1` and `b1` would have the same sequence number at that point. I was sufficiently confused that I originally wrote a comment saying that the `b1` should be changed to `txAccount`, then tested it and found to my surprise that it worked.